generate_status_quo.py

```diff
@@ -0,0 +1,336 @@
+            with contextlib.suppress(subprocess.CalledProcessError):
+                filenames = subprocess.check_output(cmd, shell=True, text=True).split('\n')
+                filenames = [x.split(':')[0] for x in filenames if x]
+                filenames = sorted(set(filenames))
+                dirnames = [os.path.dirname(x) for x in filenames]
+                dirnames = [x.replace(os.path.join(self.checkout_dir, 'lib', 'ansible', 'modules') + '/', '') for x in dirnames]
+                dirnames = sorted(set(dirnames))
+                logger.info('%s dirs' % len(dirnames))
+
+        self.orphaned = [x for x in self.pluginfiles if not x[-2]]
+
+    def make_spec(self):
+        topics = self.topics[:]
+        for idx,x in enumerate(topics):
+            if not '.' in x:
+                topics[idx] = x + '.misc'
+        self.collections['_core'] = {}
+        self.collections['_orphaned'] = []
+        for topic in topics:
+            self.collections[topic] = {}
+
+        for idx,x in enumerate(self.pluginfiles):
+            topic = x[2]
+            if topic is None:
+                self.collections['_orphaned'].append(x[-1])
+                continue
+            ptype = x[0]
+            if not '.' in topic:
+                topic = topic + '.misc'
+            if ptype not in self.collections[topic]:
+                self.collections[topic][ptype] = []
+            self.collections[topic][ptype].append(x[1])
+
+        self.collections['_orphaned'] = sorted(self.collections['_orphaned'])
+        for k,v in self.collections.items():
+            if k == '_orphaned':
+                continue
+            for ptype, pfiles in v.items():
+                self.collections[k][ptype] = sorted(pfiles)
+
+        if os.path.exists('status_quo'):
+            shutil.rmtree('status_quo')
+        os.makedirs('status_quo')
+
+        namespaces = OrderedDict()
+        for k,v in self.collections.items():
+            if '.' not in k:
+                continue
+            namespace = k.split('.')[0]
+            name = k.split('.')[1]
+            if namespace not in namespaces:
+                namespaces[namespace] = OrderedDict()
+            namespaces[namespace][name] = copy.deepcopy(v)
+
+        for namespace,names in namespaces.items():
+            fn = os.path.join('status_quo', namespace + '.yaml')
+            with open(fn, 'w') as f:
+                ruamel.yaml.dump(names, f, Dumper=ruamel.yaml.RoundTripDumper)
+
+        with open(os.path.join('status_quo', '_orphaned.yaml'), 'w') as f:
+            ruamel.yaml.dump(self.collections['_orphaned'], f, Dumper=ruamel.yaml.RoundTripDumper)
+
+
+def main():
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--filter')
+    parser.add_argument('--namespace_prefix', default=None, help='prefix each collection namespace with this string')
+    args = parser.parse_args()
+
+    StatusQuo().run()
```

Comment:
Use
```suggestion
    StatusQuo.run()
```
since it's a classmethod